be/src/exprs/vectorized/math_functions.cpp

```diff
@@ -441,20 +441,23 @@ ColumnPtr MathFunctions::conv_string(FunctionContext* context, const starrocks::
         }
 
         StringParser::ParseResult parse_res;
-        int64_t decimal_num = StringParser::string_to_int<int64_t>(reinterpret_cast<char*>(string_value.data),
+        int128_t decimal_num = StringParser::string_to_int<int128_t>(reinterpret_cast<char*>(string_value.data),
                                                                    string_value.size, src_base_value, &parse_res);
-
         if (src_base_value < 0 && decimal_num >= 0) {
             result.append_null();
             continue;
```

Comment:
Why this condition returns `null`? `decimal_num` is a signed number when `src_base_value` is negative, so maybe `decimal_num >= 0`  is legal?

Reply:
This condition seems unreasonable and will cause the results to be inconsistent with MySQL, I have fixed it @ZiheLiu